### DISCUSSION TITLE
StorageManager API - Multiple LocalStorages

### DISCUSSION THREAD

**Discussion by raywong-fb** [GrapesJS/grapesjs#4930] [Q&A] [answered]
Hi!

I'm just getting around to upgrading to the breaking Storage change that was introduced in 0.19.4. Previously, I was using the StorageManager API to add multiple storages (using localStorage). I'm having trouble figuring out exactly how to handle the new `load` function since it does not provide the `keys` parameter anymore.

Previously, my load function looked something like:
```js
StorageManager.add('storage-key', {
  load: (keys, clb) => {
    const returnObj = {};
    foreach key in keys {
      // get the respective data from local storage (localStorage.getItem(`prefix-${key}`))
      // store in returnObj
          // ie. { pages: [], assets: [] ... }
    }

    clb(returnObj);
  },
  store: (data, clb) => {
    foreach key in data {
      // store in respective local storage
      // ie. localStorage.setItem(`prefix-${key}`, data[key])
    }

    clb();
  }
});
```

I'm probably missing something simple. Any ideas on how to accomplish something similar w/ the updated StorageManager API?

**Comment by artf** [comment, marked as answer]
Yeah as you can see from the example [here](https://grapesjs.com/docs/modules/Storage.html#define-new-storage), the new storage API doesn't store project data per keys anymore but the entire JSON, that allows the `local` and `remote` to be more constant in relation how data are loaded/stored.
So if before your project would be saved in this way (eg. using the localStorage)

| Key | Value |
|--|--|
| `prefix-pages`| `{...}` |
| `prefix-style`| `{...}` |

Now it's all placed in one single value

| Key | Value |
|--|--|
| `projectKeyId`| `{ pages: ..., styles: ... }` |

**Reply by raywong-fb** [reply]
Thanks @artf! I was able to essentially use the example you linked to and just provide the prefix for multiple storages.
